### PR TITLE
ref(tsdb): fix port over non-outcomes tsdb models to use SnQL

### PR DESCRIPTION
**Pull request by barkbarkimashark**
Re-reverts and fixes https://github.com/getsentry/sentry/pull/43674

Resolves SENTRY-Y2B

### CODE REVIEW

File: src/sentry/tsdb/snuba.py [hunk 1]
```diff
@@ -322,6 +275,22 @@ def get_data(
...
+                    or_conditions = []
+                    for or_cond in cond:
+                        or_conditions.append(parse_condition(or_cond))
+
+                    if len(or_conditions) > 1:
+                        parsed_conditions.append(Or(or_conditions))
+                    else:
+                        parsed_conditions.extend(or_conditions)
+                else:
+                    parsed_conditions.append(parse_condition(cond))
```
Comment (barkbarkimashark) [inline src/sentry/tsdb/snuba.py:292]: Taken from:
https://github.com/getsentry/snuba-sdk/blob/ba9185c009ef893d0dd301c2810c1930265d60d6/snuba_sdk/legacy.py#L287-L298

Previously we weren't fully parsing the conditions correctly and left out legacy OR conditions. Not entirely sure about `AND` conditions though.

Comment (barkbarkimashark) [reply src/sentry/tsdb/snuba.py:292]: Actually, not entirely sure the absence of this bit of code was the cause of the list index out of range issue.

Comment (evanh) [reply src/sentry/tsdb/snuba.py:292]: Does your test fail if you remove this change?

Comment (barkbarkimashark) [reply src/sentry/tsdb/snuba.py:292]: Yeah, I made my original comment before adding the test. Commenting out the `if not is_condition(cond)` branch does reproduce the exception referenced in the issue.

**Review by wedamija — APPROVED** [review]
lgtm, double checked this by adding 
```
response = self.get_success_response(query="is:unresolved environment:production")
assert len(response.data) == 1
```
to https://github.com/getsentry/sentry/blob/0f0687d9644f827832e6dccdc07dc11c272314bc/tests/snuba/api/endpoints/test_organization_group_index.py#L416-L437 and removing your fix to reproduce the problem.

**Pull request merged**